sls-versions/src/main/java/com/palantir/sls/versions/NonOrderableSlsVersion.java

```diff
@@ -62,6 +62,21 @@ public static boolean check(String coordinate) {
         return safeValueOf(coordinate).isPresent() && !OrderableSlsVersion.check(coordinate);
     }
 
+    @Override
+    public final OptionalInt rcNumber() {
+        return OptionalInt.empty();
+    }
+
+    @Override
+    public final OptionalInt snapshotNumber() {
+        return OptionalInt.empty();
+    }
+
+    @Override
+    public final SlsVersionType getType() {
+        return SlsVersionType.NON_ORDERABLE;
+    }
```

Comment:
As mentioned, these introduce an ABI break on the builder type. Somehow rev-api doesn't seem to run, so I'll accept these and ensure rev-api runs in a separate PR

Same comment for `OrderableSlsVersion` below

Reply:
Seems like it was a bug in gradle-revapi, which was fixed on develop by migrating to https://github.com/revapi/gradle-revapi as part of https://github.com/palantir/sls-version-java/pull/1184